ena-submission/src/ena_deposition/upload_external_metadata_to_loculus.py

```diff
@@ -26,101 +26,202 @@
+            conditions={"status_all": status},
+        ):
+            accession = entry["accession"]
+            version = entry["version"]
+            accession_version = f"{accession}.{version}"
+            data, all_present = get_external_metadata_to_upload(db_config, entry)
+            seq_key = {"accession": accession, "version": version}
+
+            previously_uploaded: dict[str, Any] = entry.get("external_metadata", {})
+            new_external_metadata: dict[str, Any] = data.get("externalMetadata", {})
+
+            if new_external_metadata and (
+                not previously_uploaded
+                or any(
+                    previously_uploaded.get(key) != value
+                    for key, value in new_external_metadata.items()
+                )
+            ):
+                try:
+                    submit_external_metadata(
+                        data,
+                        config,
+                        entry["organism"],
+                    )
+                    logger.info(
+                        f"External metadata update for {accession_version} succeeded. "
+                        f"Old data: {previously_uploaded}, "
+                        f"new data: {new_external_metadata}"
+                    )
+                except Exception as e:
+                    logger.exception(
+                        f"Submitting external metadata to backend failed for "
+                        f"{accession_version}: {e}"
+                    )
+                    continue
+                try:
+                    update_with_retry(
+                        db_config,
+                        conditions=seq_key,
+                        update_values={
+                            "external_metadata": json.dumps(new_external_metadata),
+                        },
+                        table_name=TableName.SUBMISSION_TABLE,
+                    )
+                except Exception as e:
+                    logger.exception(
+                        f"Failed to add new state of submitted external metadata to db for "
+                        f"{accession_version}: {e}"
+                    )
+                    continue
+            if status == StatusAll.SUBMITTED_ALL and all_present:
+                try:
+                    update_with_retry(
+                        db_config,
+                        conditions=seq_key,
+                        update_values={
+                            "status_all": StatusAll.SENT_TO_LOCULUS,
+                            "finished_at": datetime.now(tz=pytz.utc),
+                        },
+                        table_name=TableName.SUBMISSION_TABLE,
+                    )
+                except Exception as e:
+                    logger.exception(
+                        f"Failed to update status_all for {accession_version} to "
+                        f"{StatusAll.SENT_TO_LOCULUS}: {e}"
+                    )
 
 
 def upload_handle_errors(
```

Comment:
your current rewrite never puts the database into this state so the error state so we never use this function and we never get a notification on slack about the issue